Manually approve pages in QA review

SuaYoo opened the pull request:
Partially addresses https://github.com/webrecorder/browsertrix-cloud/issues/1477

### Changes

- Automatically update view to first page if page ID isn't specified
- Show current page URL in location bar (resolves https://github.com/webrecorder/browsertrix-cloud/issues/1495)
- Approve, reject, or leave notes on a page
- Display temporary list of links to pages in the sidebar

You can access this view by adding `/review/screenshots` to a crawl detail page URL. The page IDs on the side are clickable.

### Manual testing

Testing steps in https://github.com/webrecorder/browsertrix-cloud/pull/1534

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| QA Review | <img width="1218" alt="Screenshot 2024-03-05 at 10 50 48 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/e4f9b1b0-9a42-424f-b091-72d8add21c92"> |
| QA Review | <img width="1104" alt="Screenshot 2024-03-11 at 3 32 21 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/b1b50ee2-fa47-4c09-8852-0c68af9dd14d"> |



<!-- ### Follow-ups -->

SuaYoo committed:
make pages request

SuaYoo committed:
show pages list

SuaYoo committed:
add location bar

SuaYoo committed:
show current page url

SuaYoo committed:
add timestamp

SuaYoo committed:
only show timestamp if present

emma-sg commented on frontend/src/pages/org/archived-item-qa.ts:
```diff
@@ -118,43 +156,144 @@ export class ArchivedItemQA extends TailwindElement {
+          ])}
+        </div>
+        <div
+          class="mx-1.5 flex h-8 flex-1 items-center justify-between rounded border bg-neutral-0 px-2 text-sm leading-none"
```
```suggestion
          class="mx-1.5 flex h-8 min-w-0 flex-1 items-center justify-between rounded border bg-neutral-0 px-2 text-sm leading-none"
```

emma-sg commented on frontend/src/pages/org/archived-item-qa.ts:
```diff
@@ -118,43 +156,144 @@ export class ArchivedItemQA extends TailwindElement {
+        <div
+          class="mx-1.5 flex h-8 flex-1 items-center justify-between rounded border bg-neutral-0 px-2 text-sm leading-none"
+        >
+          <span>${this.page?.url || "http://"}</span>
```
```suggestion
          <span class="flex-1 truncate">${this.page?.url || "http://"}</span>
```

SuaYoo replied:
Do we want to truncate the URL if it's how the current page is identified? Maybe have it scrollable instead?

emma-sg replied:
Hmm, that's a good point — should it be focusable, and switch to being scrollable when it's focused?

SuaYoo replied:
Updated to scroll!

SuaYoo replied:
Also should've noted that we'll probably revisit the location bar anyway in the subtasks, this is to add one as a starting point.

emma-sg commented on frontend/src/pages/org/archived-item-qa.ts:
```diff
@@ -118,43 +156,144 @@ export class ArchivedItemQA extends TailwindElement {
           <h1>${msg("Review")} &mdash; ${itemName}</h1>
         </header>
         <section class="main outline">
```
```suggestion
        <section class="main min-w-0 outline">
```

emma-sg reviewed (changes requested):
The toolbar doesn't seem to do well with longer URLs/smaller viewport sizes. This should probably be clipped with an ellipse? Left some suggestions that should fix this.
<img width="691" alt="image" src="https://github.com/webrecorder/browsertrix-cloud/assets/5727389/a01dcd2e-ceb7-472d-82d5-1a1ffa6f785f">

Also a couple linting issues — should be most auto-fixable, I think.

SuaYoo committed:
scroll long urls

SuaYoo committed:
Allow users to rate page in QA review (#1534)

Adds button group to QA review page for approving, rejecting, and adding
comments to a page.

---------

Co-authored-by: Henry Wilkinson <henry@wilkinson.graphics>

SuaYoo committed:
lint
